x-pack/plugin/security/qa/basic-enable-security/build.gradle

```diff
@@ -9,59 +8,53 @@ dependencies {
+    numberOfNodes = 2
     setting 'xpack.ilm.enabled', 'false'
     setting 'xpack.ml.enabled', 'false'
     setting 'xpack.license.self_generated.type', 'basic'
-    setting 'xpack.security.enabled', 'true'
-    setting 'xpack.security.authc.anonymous.roles', 'anonymous'
-    setting 'xpack.security.transport.ssl.enabled', 'true'
-    setting 'xpack.security.transport.ssl.certificate', 'transport.crt'
-    setting 'xpack.security.transport.ssl.key', 'transport.key'
-    setting 'xpack.security.transport.ssl.key_passphrase', 'transport-password'
-    setting 'xpack.security.transport.ssl.certificate_authorities', 'ca.crt'
-
-    extraConfigFile 'transport.key', project.projectDir.toPath().resolve('src/test/resources/ssl/transport.key').toFile()
-    extraConfigFile 'transport.crt', project.projectDir.toPath().resolve('src/test/resources/ssl/transport.crt').toFile()
-    extraConfigFile 'ca.crt', project.projectDir.toPath().resolve('src/test/resources/ssl/ca.crt').toFile()
+    setting 'xpack.security.enabled', 'false'
+}
 
-    setupCommand 'setupAdminUser',
-                 'bin/elasticsearch-users', 'useradd', 'admin_user', '-p', 'admin-password', '-r', 'superuser'
-    setupCommand 'setupTestUser' ,
-                 'bin/elasticsearch-users', 'useradd', 'security_test_user', '-p', 'security-test-password', '-r', 'security_test_role'
-    extraConfigFile 'roles.yml', project.projectDir.toPath().resolve('src/test/resources/roles.yml').toFile()
+task integTestSecurity(type: Test) {
+    description = "Run tests against a cluster that has security"
+    useCluster testClusters.integTest
+    dependsOn integTest
+    systemProperty 'tests.has_security', 'true'
+    maxParallelForks = 1
+    outputs.doNotCacheIf "Caching of REST tests not implemented yet", { false }
+
+    doFirst {
+        testClusters.integTest {
+            // Reconfigure cluster to enable security
+            setting 'xpack.security.enabled', 'true'
+            setting 'xpack.security.authc.anonymous.roles', 'anonymous'
+            setting 'xpack.security.transport.ssl.enabled', 'true'
+            setting 'xpack.security.transport.ssl.certificate', 'transport.crt'
+            setting 'xpack.security.transport.ssl.key', 'transport.key'
+            setting 'xpack.security.transport.ssl.key_passphrase', 'transport-password'
+            setting 'xpack.security.transport.ssl.certificate_authorities', 'ca.crt'
+
+            extraConfigFile 'transport.key', file('src/test/resources/ssl/transport.key')
+            extraConfigFile 'transport.crt', file('src/test/resources/ssl/transport.crt')
+            extraConfigFile 'ca.crt', file('src/test/resources/ssl/ca.crt')
+            extraConfigFile 'roles.yml', file('src/test/resources/roles.yml')
+
+            user username: "admin_user", password: "admin-password"
+            user username: "security_test_user", password: "security-test-password", role: "security_test_role"
+
+            restart()
```

Comment:
I believe this is the only bit that needs to be inside `doFirst { }` the rest can, and should, be done at configuration time.

Reply:
We are actually re-configuring the cluster here. 
The first test runs some tests with security disable, we then configure 
the cluster from the first test with security and test some more

Reply:
Ah I see. So we mutate the configuration here.